src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -117,17 +117,15 @@ private LambdaExpression GetProjectionLambda()
             }
         }
 
-        internal Expression CreatePropertyNameExpression(IEdmEntityType elementType, IEdmProperty property, Expression source)
+        internal Expression CreatePropertyNameExpression(IEdmStructuredType elementType, IEdmProperty property, Expression source)
         {
-            Contract.Assert(elementType != null);
             Contract.Assert(property != null);
             Contract.Assert(source != null);
 
             IEdmEntityType declaringType = property.DeclaringType as IEdmEntityType;
```

Comment:
>property.DeclaringType as IEdmEntityType; [](start = 43, length = 41)

The property maybe the structural property, the cast will give null, let's verify these codes working?